src/index.js

```diff
@@ -25,12 +25,16 @@ function useVisibility(node, options = {}) {
   const observer = new IntersectionObserver(handleObserverUpdate, options);
 
   useEffect(() => {
-    if (node.current) {
-      observer.observe(node.current);
+    const element = node.current;
```

Comment:
we have to cache it in separate variable, because the user might render a component and pass ref to element A and later rerender and pass the same ref to element B and cleanup has to be called for the previous element A